modules/registry-exposing-secure-registry-manually.adoc

```diff
@@ -15,116 +15,46 @@ images using the route host.
 ** Deploy a router.
 
 .Procedure
 
-. Verify whether a passthrough route exists, the route should have been created by
-default for the registry during the initial cluster installation:
-+
-----
-$ oc get route/pod-registry -o yaml
-apiVersion: v1
-kind: Route
-metadata:
-  name: podman-registry
-spec:
-  host: <host> <1>
-  to:
-    kind: Service
-    name: podman-registry <2>
-  tls:
-    termination: passthrough <3>
-----
-<1> The host for your route. You must be able to resolve this name externally by
-DNS to the router's IP address.
-<2> The service name for your registry.
-<3> Specifies this route as a passthrough route.
-+
-[NOTE]
-====
-Re-encrypt routes are also supported for exposing the secure registry.
-====
+. Expose the router using `DefaultRoute`, which is insecure:
 
-.. If it does not exist, create the route with the `oc create route passthrough`
-command, specifying the registry as the route’s service. By default, the name of
-the created route is the same as the service name:
+.. Set `DefaultRoute` to `True`.
 
-... Get the *podman-registry* service details:
+.. Log in insecurely with Podman:
 +
 ----
-$ oc get svc
-NAME              CLUSTER_IP       EXTERNAL_IP   PORT(S)                 SELECTOR                  AGE
-podman-registry   172.30.69.167    <none>        5000/TCP                podman-registry=default   4h
-kubernetes        172.30.0.1       <none>        443/TCP,53/UDP,53/TCP   <none>                    4h
-router            172.30.172.132   <none>        80/TCP                  router=router             4h
+$ podman login --insecure default-route.image-registry...
 ----
 
-... Create the route:
-+
-----
-$ oc create route passthrough    \
-    --service=podman-registry    \//<1>
-    --hostname=<host>
-route "podman-registry" created     <2>
-----
-<1> Specifies the registry as the route's service.
-<2> The route name is identical to the service name.
+////
 
-. Trust the certificates being used for the registry on your host
-system to allow the host to push and pull images. The certificates referenced
-were created when you secured your registry:
-+
-----
-$ sudo mkdir -p /etc/podman/certs.d/<host>
-$ sudo cp <ca_certificate_file> /etc/podman/certs.d/<host>
-$ sudo systemctl restart podman
-----
+You can expose the registry insecurely or securely:
+
+OR
 
-. Log in to the registry using the information from securing the registry.
-However, this time point to the host name used in the route rather than your
-service IP. When logging in to a secured and exposed registry, make sure you
-specify the registry in the `podman login` command:
+. Using Custom Routes, which is secure:
+
+.. Create secret:
 +
 ----
-# podman login -e user@company.com \
-    -u f83j5h6 \
-    -p Ju1PeM47R0B92Lk3AZp-bWJSck2F7aGCiZ66aFGZrs2 \
-    <host>
+$ oc create secret tls public-route-tls
+  - n image-registry\
+  -- cert=/path/to/tls.crt\
+  -- key=/path/to/tls.key\
+//  -- from-file=tls.cacrt=/path/to/tls.cacrt\
```

Comment:
From testing - passing `--from-file` won't work.

@openshift/openshift-team-network-edge I noticed that OpenShift routes let you define a `caCertificate`, but Ingress objects do not (just the tls cert and key). What is the distinction?

Reply:
Also worth noting - this part of the doc is commented out due to https://bugzilla.redhat.com/show_bug.cgi?id=1719965

Reply:
> @openshift/openshift-team-network-edge I noticed that OpenShift routes let you define a `caCertificate`, but Ingress objects do not (just the tls cert and key). What is the distinction?

The `Route` resource's `spec.tls.certificate` field specifies the server certificate whereas the `spec.tls.caCertificate` field specifies a CA certificate.  However, it is possible to create a PEM bundle that contains both the server certificate and a CA certificate and then specify this PEM bundle as the value for `spec.tls.certificate`.  That is, the `spec.tls.caCertificate` field is not really necessary to specify CA certificates.

Similar is true of the `Ingress` resource: you can create a `Secret` containing a PEM bundle that contains both the server certificate and a CA certificate and specify the name of this `Secret` as the value for an `Ingress` resource's `spec.tls[*].secretName` field.

(Technical details: OpenShift translates each `Ingress` resource and its associated `Secret` resources into one or more `Route` resources.  The contents of a `Secret` referenced by an `Ingress` resource's `spec.tls[*].secretName` field are put into a generated `Route` resource's `spec.tls.certificate` field.  Because a `Route` resource's `spec.tls.certificate` field can specify a PEM bundle, so too can an `Ingress` resource's `spec.tls[*].secretName` field.)

Reply:
@Miciah Thanks for the input. It seems in the docs the recommended procedure around TLS is to have the certificate be signed by the CA in a single PEM bundle [1]. I think we will want to continue that approach to make things consistent for admins.

I have two follow-up questions:
1. If a cluster admin sets up custom default certs on the ingress operator, will these default certs  apply to both `Ingress` objects and `Route` objects?
2. In the long run does it make sense for the registry operator to create `Ingress` objects, rather than `Route` objects?

/cc @dmage 

[1] https://docs.openshift.com/container-platform/4.1/networking/ingress-operator.html#nw-ingress-setting-a-custom-default-certificate_configuring-ingress

Reply:
> * If a cluster admin sets up custom default certs on the ingress operator, will these default certs  apply to both `Ingress` objects and `Route` objects?

Yes, because `Ingress` resources get translated into `Route` resources, and it the ingress controller (AKA router) handles all `Route` resources the same way irrespective of how they were created.

> * In the long run does it make sense for the registry operator to create `Ingress` objects, rather than `Route` objects?

We have no plans to deprecate the `Route` resource.  In general, OpenShift operators and users can use whichever resource is more convenient.  @ironcladlou, do you have anything to add to that?

(I did notice that the openshift-controller-manager operator, which starts the ingress-to-route controller, is at the same run level as the ingress operator, which means that ingress-to-route _might_ start after the ingress operator, in which case creating an `Ingress` and waiting for any corresponding `Route` resources to get created _might_ take longer than directly creating a new `Route` on a new cluster, but I don't think that's worth worrying about, and I only mention it for completeness.)

Reply:
Circling back on this - @bmcelvee you can remove the `--from-file`.

Creating the secret should be an _optional_ step - otherwise the route will use the default TLS configuration from the ingress operator.